nats/bench_command.go

```diff
@@ -147,30 +282,68 @@ func (c *benchCmd) runPublisher(bm *bench.Benchmark, nc *nats.Conn, startwg *syn
 
 	start := time.Now()
 
-	var m *nats.Msg
-	var err error
-
-	errBytes := []byte("error")
-	minusByte := byte('-')
-
-	for i := 0; i < numMsg; i++ {
-		if progress != nil {
-			progress.Incr()
-		}
-
-		if !c.ack {
-			nc.Publish(c.subject, msg)
-			continue
+	if !c.js {
+
+		var m *nats.Msg
+		var err error
+
+		errBytes := []byte("error")
+		minusByte := byte('-')
+
+		for i := 0; i < numMsg; i++ {
+			if progress != nil {
+				progress.Incr()
+			}
+
+			if !c.request {
+				err = nc.Publish(c.subject, msg)
+				if err != nil {
+					log.Fatalf("Publish error: %s", err)
+				}
+			} else {
+				m, err = nc.Request(c.subject, msg, time.Second)
+				if err != nil {
+					log.Fatalf("Request error: %s", err)
```

Comment:
should a single request failing really terminate the entire benchmark?

Reply:
Done like that by request from Colin (thought being that if the subscriber is expecting a specific number of messages it would then hang for ever (at least as a subscriber) if the publisher is unable to publish one or more of it's messages